src/routerd_lib/handlers/proxy.cpp

```diff
@@ -170,8 +294,13 @@ namespace NAC {
         bool contentDispositionFormData
     ) const {
         ServiceReplied(request, serviceName);
-
-        if ((serviceName == std::string("output")) && !request->IsResponseSent()) { // TODO
+        if (
+            (serviceName == std::string("output")
+             || (Graph.Services.count(serviceName) > 0
+                 && Graph.Services.at(serviceName).SaveAs == std::string("output"))
```

Comment:
This is incredibly kludgy and unnecessary in general. Just call `ProcessServiceResponse` with `service.Save` as part name in `Iter` in that case. And do not forget to call `ServiceReplied` with original service name after that, the same way as in multipart/x-ac-routerd case.